### PR TITLE
[Misc] Allow Meloetta's starting form to be selected

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes the user will see?
Players can now choose between Meloetta's different forms on the starter select screen.
Additionally, Meloetta will not change forms during a mono-type challenge when using Relic Song nor automatically revert to Aria form on arena reset.

## Why am I making these changes?
image
image
image

## What are the changes from a developer perspective?
Marked the second form as selectable in `pokemon-species.ts`.
Created `MeloettaFormChangePostMoveTrigger` which extends `SpeciesFormChangePostMoveTrigger` and adds a check for whether a mono-type challenge is currently active.
Removed `SpeciesFormChangeActiveTrigger(false)` which caused Meloetta to revert to Aria form automatically.

### Screenshots/Videos
image
image

## How to test the changes?
Go to the starter select screen and try to change Meloetta's form.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- ~[ ] If I have text, did I make it translatable and add a key in the English locale file(s)?~
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by damocleas — APPROVED** [review]

File: src/data/pokemon-forms.ts [hunk 1]
```diff
@@ -759,9 +770,8 @@ export const pokemonFormChanges: PokemonFormChanges = {
     new SpeciesFormChange(Species.KELDEO, "resolute", "ordinary", new SpeciesFormChangeMoveLearnedTrigger(Moves.SECRET_SWORD, false))
   ],
   [Species.MELOETTA]: [
-    new SpeciesFormChange(Species.MELOETTA, "aria", "pirouette", new SpeciesFormChangePostMoveTrigger(Moves.RELIC_SONG), true),
-    new SpeciesFormChange(Species.MELOETTA, "pirouette", "aria", new SpeciesFormChangePostMoveTrigger(Moves.RELIC_SONG), true),
-    new SpeciesFormChange(Species.MELOETTA, "pirouette", "aria", new SpeciesFormChangeActiveTrigger(false), true)
```
Comment (torranx) [inline src/data/pokemon-forms.ts:764]: Why was this removed?

Comment (innerthunder) [reply src/data/pokemon-forms.ts:764]: This reverts Pirouette form to Aria form when Meloetta's removed from the field. Balance team wanted this to be removed a while ago, but it broke monotype challenges without the `MeloetteFormChangePostMoveTrigger` stuff that's included in this PR

**Review by innerthunder — APPROVED** [review]

File: src/data/pokemon-forms.ts (hunk 1 above)
Comment (torranx) [reply src/data/pokemon-forms.ts:764]: Ah, so we intentionally want to it to stay in form. Oks.

**Review by torranx — APPROVED** [review]
would be better if the form-change class has docs in it but otherwise looks good
